src/main/kotlin/io/github/gradlenexus/publishplugin/internal/ActionRetrier.kt

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.github.gradlenexus.publishplugin.internal
+
+interface ActionRetrier<R> {
+    fun execute(operationToExecuteWithRetrying: () -> R): R
+}
```

Comment:
Kotlin does not support implementation of `Kotlin` interfaces via lambdas.

Do you think it would be better to use `typealias ActionRetrier`?

Reply:
Ah, `typealias` can't have generics :(

Reply:
> Kotlin does not support implementation of `Kotlin` interfaces via lambdas.

I didn't know. I could change it to Java interface if that helped.

> Do you think it would be better to use `typealias ActionRetrier`?

`typealias` seems to be more an a type alias (like in Groovy or Scala) and I would like to have an interface with (potentially) multiple implementations.

Reply:
>I could change it to Java interface if that helped.

In fact, it would help. In other words, Kotlin can implement Java interfaces via lambdas.
However, it is a bit weird for me :(

Reply:
I gave it a try, but it complicates the current implementation a little bit.

```
interface ActionRetrier<R> {
    R execute(Supplier<R> operationToExecuteWithRetrying);
}
```
cannot be used as FailSafe which we use expects its own:
```
public interface CheckedSupplier<T> {
  T get() throws Exception;
}
```
Callable would be ok, but at the type level they are not compatible. As a result:
```
    override fun execute(operationToExecuteWithRetrying: Callable<R>): R {
        return Failsafe.with(retrier).get(
            { operationToExecuteWithRetrying.call() } as CheckedSupplier<R>) }
        )
```
is required which doesn't look good. 

I will merge it as is (with the Kotlin interface). Feel free to create a separate PR to improve that.